src/main/java/com/puppycrawl/tools/checkstyle/checks/naming/MethodNameCheck.java

```diff
@@ -39,39 +39,32 @@
  * <ul>
  * <li>
  * Property {@code allowClassName} - Controls whether to allow a method name to have the same name
- * as the residing class name. This is not to be confused with a constructor. An easy mistake is
- * to place a return type on a constructor declaration which turns it into a method. For example:
- * <pre>
- * class MyClass {
- *     public void MyClass() {} //this is a method
- *     public MyClass() {} //this is a constructor
- * }
- * </pre>
+ * as the residing class name. This is not to be confused with a constructor.
```

Comment:
this was required, as our macros does not generate well formatted xml for embeded code block.
after reading attentively, I do think we need this example.